samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,579 @@
+	return 0;
+
+error:
+	zsock_close(ctx->sock);
+	ctx->sock = -1;
+	zsock_freeaddrinfo(res);
+	return -errno;
+}
+
+/*
+ * @brief HTTP response callback - processes firmware chunks
+ */
+static int ota_http_response_cb(struct http_response *rsp, enum http_final_call final_data,
+				void *user_data)
+{
+	ARG_UNUSED(final_data);
+	struct app_ctx *ctx = (struct app_ctx *)user_data;
+
+	ctx->http_status_code = rsp->http_status_code;
+	ctx->image_size = rsp->content_range.total;
+
+	if (rsp->body_frag_len > 0) {
+		if (ctx->flash_buffer_len + rsp->body_frag_len > sizeof(ctx->flash_buffer)) {
+			return -ENODATA;
+		}
+		memcpy(ctx->flash_buffer + ctx->flash_buffer_len, rsp->body_frag_start,
+		       rsp->body_frag_len);
+		ctx->flash_buffer_len += rsp->body_frag_len;
+	}
+
+	return 0;
+}
+
+/*
+ * @brief Download firmware chunk from server
+ */
+static int ota_download_firmware_chunk(struct app_ctx *ctx)
+{
+	char range_header[64];
+	const char *headers[] = {range_header, NULL};
+	int ret = 0;
+	struct http_request req = {
+		.method = HTTP_GET,
+		.url = ctx->url.path,
+		.header_fields = headers,
+		.host = ctx->url.host,
+		.protocol = "HTTP/1.1",
+		.response = ota_http_response_cb,
+		.recv_buf = ctx->http_recv_buffer,
+		.recv_buf_len = sizeof(ctx->http_recv_buffer)
+	};
+
+	ctx->flash_buffer_len = 0;
+
+	/* Prepare headers */
+	snprintf(range_header, sizeof(range_header), "Range: bytes=%d-%d\r\n", ctx->range_start,
+		 ctx->range_start + sizeof(ctx->flash_buffer) - 1);
+
+	/* Send request */
+	ret = http_client_req(ctx->sock, &req, 3000, ctx);
+	if (ret < 0) {
+		printf("HTTP request failed: %s, requested range=%d-%d\n", strerror(-ret),
+		       ctx->range_start, ctx->range_start + sizeof(ctx->flash_buffer) - 1);
+		return ret;
+	}
+
+	/* Handle HTTP status codes - centralized error management */
+	if (ctx->http_status_code >= 300) {
+		printf("HTTP error %d, requested range=%d-%d\n", ctx->http_status_code,
+		       ctx->range_start, ctx->range_start + sizeof(ctx->flash_buffer) - 1);
+		return -EPROTO;
+	}
+	return ret;
+}
+
+static int ota_update_http_range_header(struct app_ctx *ctx)
+{
+	ctx->range_start += sizeof(ctx->flash_buffer);
+
+	if (ctx->range_start >= (int)ctx->image_size) {
+		return -ENODATA;
+	}
+	return 0;
+}
+
+/*
+ * @brief Process firmware update data packets
+ */
+static int ota_load_firmware(struct app_ctx *ctx)
+{
+	uint32_t ota_calc_image_size = 0;
+	uint32_t status = 0;
+	int buf_offset = 0;
+
+	__ASSERT(ctx->flash_buffer_len > 0, "No firmware data to process");
+
+	if (ctx->range_start == 0) {
+		buf_offset = sizeof(sl_si91x_firmware_header_t);
+		status = sl_wifi_get_firmware_size((void *)ctx->flash_buffer, &ota_calc_image_size);
+		if (status != SL_STATUS_OK) {
+			printf("Unable to fetch firmware size. Status: 0x%x\n", status);
+			return -EIO;
+		}
+
+		if (ctx->image_size != ota_calc_image_size) {
+			printf("Image size mismatch - expected: %u, calculated: %u (file may be "
+			       "corrupted)\n",
```

Comment:
Some user may understand `file may be corrupted` means the device is now corrupted. I suggest `(corrupted payload?)` (or `Corrupted payload (image size mismatch: %d bytes != %d bytes)\n`)